### PR TITLE
[5.1] Convert mod_version to service provider

### PR DESCRIPTION
**Pull request by joomlaweby**
### Summary of Changes

Converts the version administrator module to service provider.

### Testing Instructions

1. Make sure you have administrator module mod_version published
2. You should see module output

mod_messages

### Actual result BEFORE applying this Pull Request

Module works as expected

### Expected result AFTER applying this Pull Request

Module works same way as before applying pull request

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [X] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [X] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Comment by fgsw** [comment]
I have tested this item :white_check_mark: successfully on 366e8848e2d866be710c85bd59568ad70f047b2d<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42814">issues.joomla.org/tracker/joomla-cms/42814</a>.</sub>

File: administrator/modules/mod_version/src/Dispatcher/Dispatcher.php [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+// phpcs:disable PSR1.Files.SideEffects
+\defined('_JEXEC') or die;
+// phpcs:enable PSR1.Files.SideEffects
+
+/**
+ * Dispatcher class for mod_version
+ *
+ * @since  __DEPLOY_VERSION__
+ */
+
```
Comment (sandewt) [inline administrator/modules/mod_version/src/Dispatcher/Dispatcher.php]: Please delete this empty line (26)

Comment (joomlaweby) [reply administrator/modules/mod_version/src/Dispatcher/Dispatcher.php]: Fixed

File: administrator/modules/mod_version/src/Helper/VersionHelper.php [hunk 2]
```diff
@@ -21,17 +21,35 @@
...
+     *
+     * @return  string  String containing the current Joomla version.
+     *
+     * @deprecated __DEPLOY_VERSION__ will be removed in 6.0
+     *             Use the non-static method getVersionString
+     *             Example: Factory::getApplication()->bootModule('mod_version', 'administrator')
+     *                            ->getHelper('VersionHelper')
+     *                            ->getVersionString()
+     */
+    public function getVersion()
```
Comment (sandewt) [inline administrator/modules/mod_version/src/Helper/VersionHelper.php]:
```suggestion
    public static function getVersion()
```

Comment (joomlaweby) [reply administrator/modules/mod_version/src/Helper/VersionHelper.php]: Fixed

File: administrator/modules/mod_version/services/provider.php [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+/**
+ * @package     Joomla.Administrator
+ * @subpackage  mod_version
+ *
+ * @copyright   (C) 2024 Open Source Matters, Inc. <https://www.joomla.org>
+ * @license     GNU General Public License version 2 or later; see LICENSE.txt
+ */
+
+defined('_JEXEC') or die;
```
Comment (richard67) [inline administrator/modules/mod_version/services/provider.php]:
```suggestion
\defined('_JEXEC') or die;
```
Same as in the other PR #42827 : New code style rules in 5.0-dev and 5.1-dev. This suggestion will fix PHPCS failing in Drone.

Comment (joomlaweby) [reply administrator/modules/mod_version/services/provider.php]: Fixed

**Comment by richard67** [comment]
@fgsw Could you test this PR again? Thanks in advance.

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42814">issues.joomla.org/tracker/joomla-cms/42814</a>.</sub>

File: administrator/modules/mod_version/services/provider.php [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+/**
+ * @package     Joomla.Administrator
+ * @subpackage  mod_version
+ *
+ * @copyright   (C) 2024 Open Source Matters, Inc. <https://www.joomla.org>
+ * @license     GNU General Public License version 2 or later; see LICENSE.txt
+ */
+
+ \defined('_JEXEC') or die;
```
Comment (richard67) [inline administrator/modules/mod_version/services/provider.php]:
```suggestion
\defined('_JEXEC') or die;
```
Fix PHPCS failing in drone due to space at beginning of this line.

**Comment by LadySolveig** [comment]
Thank you @joomlaweby for your work and your endurance! 
I am of course also very very happy and grateful for the tests @fgsw @ceford and the cs-fix @richard67

**Pull request merged**